pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go

```diff
@@ -96,74 +110,94 @@ func (r *HTTPRouteReconciler) gapiToKumaRoute(ctx context.Context, mesh string,
 		selectors = append(selectors, &mesh_proto.Selector{
 			Match: match,
 		})
-	}
 
-	var hostnames []string
+		refRoutes[ref] = &mesh_proto.GatewayRoute{Selectors: selectors}
+	}
 
-	for _, hn := range route.Spec.Hostnames {
-		hostnames = append(hostnames, string(hn))
+	routeConf, routeParentStatus, err := r.gapiToKumaRouteConf(ctx, mesh, route)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	var rules []*mesh_proto.GatewayRoute_HttpRoute_Rule
+	var routes []*mesh_proto.GatewayRoute
+	status := ParentStatuses{}
+
+	for ref, route := range refRoutes {
+		if routeConf != nil {
+			route.Conf = routeConf
+			routes = append(routes, route)
+		}
+
+		status[ref] = gatewayapi.RouteParentStatus{
+			ParentRef:      ref,
+			ControllerName: controllerName,
+			Conditions:     routeParentStatus,
+		}
+	}
 
-	for _, rule := range route.Spec.Rules {
-		var backends []*mesh_proto.GatewayRoute_Backend
+	return routes, status, nil
+}
 
-		for _, backend := range rule.BackendRefs {
-			ref := backend.BackendObjectReference
+func (r *HTTPRouteReconciler) shouldHandleParentRef(
+	ctx context.Context, route kube_client.Object, ref gatewayapi.ParentRef,
+) (bool, error) {
+	namespace := route.GetNamespace()
+	if ns := ref.Namespace; ns != nil {
+		namespace = string(*ns)
```

Comment:
FWIW, when we cross namespaces, we also need to look at the allowed routes on the relevant listener

https://github.com/kumahq/kuma/issues/3624